libbeat/processors/actions/decode_base64_field.go

```diff
@@ -0,0 +1,111 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package actions
+
+import (
+	"encoding/base64"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/processors"
+	"github.com/elastic/beats/libbeat/processors/checks"
+)
+
+const (
+	processorName = "decode_base64_field"
+)
+
+type decodeBase64Fields struct {
+	log *logp.Logger
+
+	field  string
+	target *string
+}
+
+type base64Config struct {
+	Field  string  `config:"field"`
+	Target *string `config:"target"`
+}
+
+var (
+	defaultBase64Config = base64Config{}
+)
+
+func init() {
+	processors.RegisterPlugin(processorName,
+		checks.ConfigChecked(NewDecodeBase64Field,
+			checks.RequireFields("field"),
+			checks.AllowedFields("field", "target", "when")))
+}
+
+// NewDecodeBase64Field construct a new decode_base64_field processor.
+func NewDecodeBase64Field(c *common.Config) (processors.Processor, error) {
+	config := defaultBase64Config
+
+	log := logp.NewLogger(processorName)
+
+	err := c.Unpack(&config)
+	if err != nil {
+		return nil, fmt.Errorf("fail to unpack the %s configuration: %s", processorName, err)
+	}
+
+	return &decodeBase64Fields{
+		log:    log,
+		field:  config.Field,
+		target: config.Target,
+	}, nil
+}
+
+func (f *decodeBase64Fields) Run(event *beat.Event) (*beat.Event, error) {
+	data, err := event.GetValue(f.field)
+	if err != nil && errors.Cause(err) != common.ErrKeyNotFound {
```

Comment:
I think it would be good to make this configurable, just like in case of other processors. By adding `fail_on_error` and `ignore_missing` users could eliminate flooding the logs of Beats if the keys are not present in the event. See the implementation in `rename` processor: https://github.com/elastic/beats/blob/master/libbeat/processors/actions/rename.go#L36